instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/internal/JdbcConnectionUrlParser.java

```diff
@@ -831,6 +831,81 @@ DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
 
       return MODIFIED_URL_LIKE.doParse(jdbcUrl, builder);
     }
+  },
+  INFORMIX_SQLI("informix-sqli") {
+    @Override
+    DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
+      builder = MODIFIED_URL_LIKE.doParse(jdbcUrl, builder);
+      builder = INFORMIX.doParse(jdbcUrl, builder);
+
+      int hostIndex = jdbcUrl.indexOf("://");
+      if (hostIndex == -1) {
+        return builder;
+      }
+
+      int dbNameStartIndex = jdbcUrl.indexOf('/', hostIndex + 3);
+      if (dbNameStartIndex == -1) {
+        return builder;
+      }
+      int dbNameEndIndex = jdbcUrl.indexOf(':', dbNameStartIndex);
+      if (dbNameEndIndex == -1) {
+        dbNameEndIndex = jdbcUrl.length();
+      }
+      String name = jdbcUrl.substring(dbNameStartIndex + 1, dbNameEndIndex);
+      if (name.isEmpty()) {
+        builder.name(null);
+      } else {
+        builder.name(name);
+      }
+
+      return builder;
+    }
+  },
+
+  INFORMIX_DIRECT("informix-direct") {
+    private static final String DEFAULT_HOST = "infxhost";
+
+    @Override
+    DbInfo.Builder doParse(String jdbcUrl, DbInfo.Builder builder) {
+      builder = MODIFIED_URL_LIKE.doParse(jdbcUrl, builder);
+      builder = INFORMIX.doParse(jdbcUrl, builder);
+      builder.host(DEFAULT_HOST);
+
+      int hostIndex = jdbcUrl.indexOf("://");
+      if (hostIndex == -1) {
+        return builder;
+      }
+
+      int informixUrlStartIdx =
+          jdbcUrl.indexOf("informix-direct://") + "informix-direct://".length();
+      int colonLoc = jdbcUrl.indexOf(":", informixUrlStartIdx);
```

Comment:
looking at https://www.ibm.com/docs/en/informix-servers/12.10?topic=method-format-database-urls it seems that `;` could also be a separator